rcp/org.testeditor.rcp4/src/org/testeditor/rcp4/dialogs/NetworkConnectionSettingDialog.java

```diff
@@ -211,6 +237,58 @@ public void widgetSelected(SelectionEvent e) {
 		});
 	}
 
+	protected void extractProxyFromMavenSettings(File mavenSettingsFile) {
+		try {
+			if (mavenSettingsFile.exists()) {
+				Document doc = DocumentBuilderFactory.newInstance().newDocumentBuilder().parse(mavenSettingsFile);
+				NodeList nodeList = doc.getElementsByTagName("proxy");
+				if (nodeList.getLength() > 0) {
+					Node item = nodeList.item(0);
+					NodeList childNodes = item.getChildNodes();
+					for (int i = 0; i < childNodes.getLength(); i++) {
+						if (childNodes.item(i).getNodeName().equalsIgnoreCase("host")) {
+							proxyHostSetting = childNodes.item(i).getTextContent();
+							updateModelAndUIFrom(proxyHost, proxyHostSetting);
+						}
+						if (childNodes.item(i).getNodeName().equalsIgnoreCase("nonProxyHosts")) {
+							noProxyHostsSetting = childNodes.item(i).getTextContent();
+							updateModelAndUIFrom(noProxyHosts, noProxyHostsSetting);
+						}
+						if (childNodes.item(i).getNodeName().equalsIgnoreCase("port")) {
+							proxyPortSetting = childNodes.item(i).getTextContent();
+							updateModelAndUIFrom(proxyPort, proxyPortSetting);
+						}
+						if (childNodes.item(i).getNodeName().equalsIgnoreCase("username")) {
+							proxyUserSetting = childNodes.item(i).getTextContent();
+							updateModelAndUIFrom(proxyUser, proxyUserSetting);
+						}
+						if (childNodes.item(i).getNodeName().equalsIgnoreCase("password")) {
+							proxyPwdSetting = childNodes.item(i).getTextContent();
+							updateModelAndUIFrom(proxyPwd, proxyPwdSetting);
+						}
+					}
+				}
+			}
+		} catch (SAXException | IOException | ParserConfigurationException e) {
+			logger.error("Error reading maven settings.", e);
+		}
+	}
+
+	protected void updateModelAndUIFrom(Text text, String textContent) {
```

Comment:
Was this method supposed to update an internal model? Seems like it only updates the UI - in this case we should use a different name here.